nixos/modules/config/nix.nix

```diff
@@ -332,6 +332,16 @@ in
                 always allowed to connect.
               '';
             };
+
+            flake-registry = mkOption {
+              type = types.nullOr types.str;
+              default = "https://channels.nixos.org/flake-registry.json";
+              description = lib.mdDoc ''
+                Path or URI of the global flake registry.
+
+                When empty, disables the global flake registry.
+              '';
+            };
```

Comment:
Infuriating bug probably introduced by Nix itself recently: the config checker complains that `flake-registry` is set if flakes aren't enabled. This is pants at the best of times and more to the point, this would have to be special-cased in the config builder in nixpkgs to ignore it when it's the default. ughhhhh

Reply:
Filed a Nix bug https://github.com/NixOS/nix/issues/9337

Reply:
Problem solved by deleting the `flake-registry` structured config, I suppose.